test/unit/plugins.test.js

```diff
@@ -1,214 +0,0 @@
-
-  const comp = player.getChild('component');
-
-  assert.ok(comp, 'Plugin added a component to the player');
-
-  player.dispose();
-});
-
-QUnit.test('Plugin should overwrite plugin of same name', function(assert) {
-  let v1Called = 0;
-  let v2Called = 0;
-  let v3Called = 0;
-
-  // Create initial plugin
-  registerPlugin('myPlugin5', function(options) {
-    v1Called++;
-  });
-  const player = TestHelpers.makePlayer({});
-
-  player.myPlugin5({});
-
-  // Overwrite and create new player
-  registerPlugin('myPlugin5', function(options) {
-    v2Called++;
-  });
-  const player2 = TestHelpers.makePlayer({});
-
-  player2.myPlugin5({});
-
-  // Overwrite and init new version on existing player
-  registerPlugin('myPlugin5', function(options) {
-    v3Called++;
-  });
-  player2.myPlugin5({});
-
-  assert.ok(v1Called === 1, 'First version of plugin called once');
-  assert.ok(v2Called === 1, 'Plugin overwritten for new player');
-  assert.ok(v3Called === 1, 'Plugin overwritten for existing player');
-
-  player.dispose();
-  player2.dispose();
-});
-
-QUnit.test('Plugins should get events in registration order', function(assert) {
-  const order = [];
-  const expectedOrder = [];
-  const pluginName = 'orderPlugin';
-  const player = TestHelpers.makePlayer({});
-  const plugin = function(name) {
-    registerPlugin(name, function(opts) {
-      this.on('test', function(event) {
-        order.push(name);
-      });
-    });
-    player[name]({});
-  };
-
-  for (let i = 0; i < 3; i++) {
-    const name = pluginName + i;
-
-    expectedOrder.push(name);
-    plugin(name);
-  }
-
-  registerPlugin('testerPlugin', function(opts) {
-    this.trigger('test');
-  });
-
-  player.testerPlugin({});
-
-  assert.deepEqual(order,
-            expectedOrder,
-            'plugins should receive events in order of initialization');
-  player.dispose();
-});
-
-QUnit.test('Plugins should not get events after stopImmediatePropagation is called', function(assert) {
-  const order = [];
-  const expectedOrder = [];
-  const pluginName = 'orderPlugin';
-  const player = TestHelpers.makePlayer({});
-  const plugin = function(name) {
-    registerPlugin(name, function(opts) {
-      this.on('test', function(event) {
-        order.push(name);
-        event.stopImmediatePropagation();
-      });
-    });
-    player[name]({});
-  };
-
-  for (let i = 0; i < 3; i++) {
-    const name = pluginName + i;
-
-    expectedOrder.push(name);
-    plugin(name);
-  }
-
-  registerPlugin('testerPlugin', function(opts) {
-    this.trigger('test');
-  });
-
-  player.testerPlugin({});
-
-  assert.deepEqual(order,
-            expectedOrder.slice(0, order.length),
-            'plugins should receive events in order of ' +
-            'initialization, until stopImmediatePropagation');
-
-  assert.equal(order.length, 1, 'only one event listener should have triggered');
-  player.dispose();
-});
-
-QUnit.test('Plugin that does not exist logs an error', function(assert) {
```

Comment:
This is now tested in player.test.js, under the "options: plugins" test. I've changed so it throws now instead of just logging. Failing loud in this case seems like a good idea.